### ISSUE TITLE
maximum session duration exceeded immediately after login

### ISSUE THREAD

**Issue by zandbelt** [pingidentity/mod_auth_openidc#193]
When compiling the module from source, on certain platforms, with certain compilers and certain webserver environments - i.e. Centos 6.4 64 bit with Oracle HTTP Server 12c (12.1.3) with Oracle specific modules installed - the conversion of timestamps to strings using APR time formatting routines may not work which would result in session timeouts being set to 0 and sessions being expired right away.

The following message showing immediately after first login would indicate this:
`oidc_check_max_session_duration: maximum session duration exceeded for user: <username>` 

It seems due to a conversion issue with `APR_TIME_T_FMT` as in:
https://github.com/pingidentity/mod_auth_openidc/blob/v2.1.0/src/mod_auth_openidc.c#L889

`

**Comment by zandbelt** [comment]
more details on the environment with probems:

```
[root]$ yum list installed|grep apr
apr.x86_64                            1.3.9-5.el6_2
apr-devel.x86_64                      1.3.9-5.el6_2
apr-util.x86_64                       1.3.9-3.el6_0.1
apr-util-devel.x86_64                 1.3.9-3.el6_0.1
apr-util-ldap.x86_64                  1.3.9-3.el6_0.1
 
[root]$ yum list installed | grep -i gcc
gcc.x86_64                            4.4.7-4.el6
gcc-c++.x86_64                        4.4.7-4.el6
libgcc.i686                           4.4.7-4.el6
libgcc.x86_64                         4.4.7-4.el6
```

**Comment by zandbelt** [comment]
I am not able to reproduce this anywhere so I blame it on an anomaly in the compilation environment.

**Comment by zandbelt** [comment]
this was fixed in https://github.com/zmartzone/mod_auth_openidc/commit/99ccd3ca64eb1ea69054c011a95dac16ff16d6cc and included in 2.4.8.2 and upwards
